code/framework/extension/src/main/java/io/cattle/platform/extension/ExtensionManager.java

```diff
@@ -10,6 +11,8 @@
 
     List<?> list(String key);
 
+    Map<String, Object> map(String key);
```

Comment:
Oh wait, I think misunderstood what this code is doing.  What about `<T> Map<String,T> map(String key)`